services/worker-manager/src/providers/aws.js

```diff
@@ -298,14 +299,20 @@ class AwsProvider extends Provider {
       workerId: worker.workerId,
     });
     monitor.debug('setting state to RUNNING');
+    const secret = `${slug.nice()}${slug.nice()}`;
     await worker.update(this.db, worker => {
       worker.lastModified = new Date();
       worker.providerData.terminateAfter = expires.getTime();
       worker.state = Worker.states.RUNNING;
+      worker.secret = this.db.encrypt({ value: Buffer.from(secret, 'utf8') });
```

Comment:
The secret for *re*registration is generated in `api.js`.  Unless there's some reason for the functionality to differ per provider, I think it will make sense to do the same for registration, rather than implementing it in each provider.